### PR TITLE
test: ITStorage converted to ITObjectTest

### PR DESCRIPTION
**Pull request by sydney-munro**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java [hunk 1]
```diff
@@ -1069,28 +1008,6 @@ public void testCopyBlobFail() {
     }
   }
 
-  @Test
-  public void testReadAndWriteChannels() throws IOException {
```
Comment (sydney-munro) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java:1073]: I feel like this is covered via the various other Read/Write channel tests

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java [hunk 2]
```diff
@@ -1069,28 +1008,6 @@ public void testCopyBlobFail() {
...
-      readStringBytes = ByteBuffer.allocate(stringBytes.length);
-      reader.read(readBytes);
-      reader.read(readStringBytes);
-    }
-    assertArrayEquals(BLOB_BYTE_CONTENT, readBytes.array());
-    assertEquals(BLOB_STRING_CONTENT, new String(readStringBytes.array(), UTF_8));
-  }
-
   @Test
   public void testReadAndWriteChannelWithEncryptionKey() throws IOException {
```
Comment (sydney-munro) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java:1012]: Asking for opinions here, there are 3 tests that don't seem to fit nicely into other Read/Write channel tests to move.
- testReadAndWriteChannelWithEncryptionKey
- testReadAndWriteChannelsWithDifferentFileSize
- testReadAndWriteCaptureChannels

Any opinions on what to do with these tests?

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITDownloadBlobWithoutAuth.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  private static String bucketName;
+
+  @BeforeClass
+  public static void setup() {
+    storage = storageFixture.getInstance();
+    bucketName = bucketFixture.getBucketInfo().getName();
+  }
+
+  @Test
+  public void testDownloadPublicBlobWithoutAuthentication() {
```
Comment (BenWhitehead) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITDownloadBlobWithoutAuth.java:57]: nit: this is three tests: 1) list, 2) download, 3) create. It'd be nice to split these up, can be done in a later PR.

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java (hunk 1 above)
Comment (BenWhitehead) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java:1073]: Agreed, this isn't doing anything unique.

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java (hunk 2 above)
Comment (BenWhitehead) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java:1012]: * `testReadAndWriteChannelWithEncryptionKey` I think is okay to remove, we now have other tests which actually verify args like encryption key are wired through.
* `testReadAndWriteChannelsWithDifferentFileSize` semi useful, but written in such a way that it's painful. We can refactor this one at a later point.
* `testReadAndWriteCaptureChannels` unfortunately we need to keep this, this is verifying that a "captured" channel can be restored - despite the fact that it's often confusing.

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITWriteChannelConnectionPoolTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import com.google.cloud.storage.testing.RemoteStorageHelper;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class ITWriteChannelConnectionPoolTest {
```
Comment (BenWhitehead) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITWriteChannelConnectionPoolTest.java:41]: IMO this test isn't valuable for us. It's strictly changing values on an api service we don't have direct control or consumption of. It's also not actually validating any pooling happens. We can remove this class.

**Review by BenWhitehead — APPROVED** [review]
Some small things we can cleanup, but feel free to merge after deciding to do in this PR or another one.

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITWriteChannelConnectionPoolTest.java (hunk 4 above)
Comment (frankyn) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITWriteChannelConnectionPoolTest.java:41]: The original reason I wrote this was due to a customer issue where the http body wasn't read / cleared and left connections pending when using connection pooling.

We can most likely remove it at this point; it was mainly used to validate a fix that was made at that time.

**Pull request merged**